### PR TITLE
Refactor(handler): Enhanced Command Handling, Basic Commands, and Modularized Client Class

### PR DESCRIPTION
**Pull request by Chere3**
### Enhanced Command Handling, Basic Commands, and Modularized Client Class
---

This draft PR addresses significant improvements in command handling, the introduction of basic commands, and the replacement of the default client class with a modularized custom client class for better organization and extendability, as outlined in the project modules *[MO-16](https://fate-pisces-a9b.notion.site/Reestructuraci-n-del-handler-8c6fc5b717094064b9b19a5724a9fdd3)* and *[MO-15](https://fate-pisces-a9b.notion.site/Comandos-b-sicos-f3dd60d110784513bc35bdff42c05ebd)* on the project roadmap. The key improvements and additions include:

- [x] **Command Context Enhancement:** Enhance the handler to include command context, providing a structured and consistent approach to access relevant command information during execution.
  
- [x] **Command Retrieval Methodology:** Revise the existing methods for retrieving commands to ensure accuracy and efficiency.
  
- [x] **Basic Command Implementation:** Implement essential commands such as `eval`, `help`, and `ping` to enrich the bot's interactive capabilities.
  
- [x] **Modularized Custom Client Class:** Replace the default client class with a custom, modularized client class to better organize the code structure and facilitate extendability for future features and commands.

These changes aim to streamline the command processing workflow, expand the bot's functionality with fundamental commands, and establish a more organized and scalable codebase through a custom client class. The implementation aligns with the outlined objectives in the project roadmap, laying a solid foundation for subsequent enhancements in the command handling mechanism and the addition of more complex commands.

---

### CODE REVIEW

**Review by Arkirion — COMMENTED** [review]
image
 folder structure regardinf functions are still weird/not intuitive, maybe we can put them inside a folder called "functions"? we can discuss this on discord.
 
 PD: Maybe since we are doing this only for events we could create the folder inside "events". should it break event loader?

File: src/modules/bot/client.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  commands: Collection<string, SlashCommand> = new Collection<
+    string,
+    SlashCommand
+  >();
+  cooldowns: Collection<string, Collection<string, number>> = new Collection<
+    string,
+    Collection<string, number>
+  >();
+
+  public async login(token?: string | undefined): Promise<string> {
```
[the remark this answers is not in the document]
Comment (Chere3) [reply src/modules/bot/client.ts:35]: Discord.js inference types:
https://old.discordjs.dev/#/docs/discord.js/main/class/Client?scrollTo=login
image

**Review by Alucherdi — APPROVED** [review]
TODO: Check for discord types to avoid infering in handler function parameters

**Comment by Chere3** [comment]
> TODO: Check for discord types to avoid infering in handler function parameters

Resolved.

**Pull request merged**